opencog/atoms/core/Variables.cc

```diff
@@ -278,6 +278,25 @@ Handle FreeVariables::substitute_scoped(const Handle& term,
 		{
 			// Make a copy... this is what's computationally expensive.
 			IndexMap hidden_map = index_map;
+
+			// Remove from hidden_map all non variables in case they
+			// contain hidden variables.
+			for (auto it = hidden_map.begin(); it != hidden_map.end();)
+			{
+				if (it->first->get_type() != VARIABLE_NODE)
```

Comment:
Should almost surely include GLOB_NODE.

Reply:
I'm confused by your fix. It is immediately above a block of code that says "Remove the alpha-hidden variables.", which also loops .. and removes variables. ... from `hidden_map`. So you added a second loop that also does "the same thing" as the first loop.  Except, clearly, it's not quite exactly "the same thing", because it fixes your bug.   .. which means that the original loop was somehow buggy. Or something.     This is not quite right. There should not be two loops that do the same thing.  (see lines 301-308 in the new file).

Reply:
oh.. I see what is going on. OK, disregard previous commment.